# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

<div class="span5 alert alert-info">
### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet
</div>
****

In [1]:
import pandas as pd
import numpy as np
from scipy import stats

In [2]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [3]:
# number of callbacks for black-sounding names
sum(data[data.race=='b'].call)

157.0

In [4]:
Black_calls = data[data.race=='b'][['race', 'call']]
White_calls = data[data.race=='w'][['race', 'call']]

In [5]:
data.head()

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...          1.0     0.0    1.0       0.0       0.0   0.0   
1            316    ...          1.0     0.0    1.0       0.0       0.0   0.0   
2             19    ...          1.0     0.0    1.0       0.0       0.0   0.0   
3            313    ...          1.0     0.0    1.0       0.0       0.0   0.0   
4            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  

[5 rows x 65 columns]

In [18]:
data.shape

(4870, 65)

# 1
What test is appropriate for this problem? Does CLT apply?

# Solution:
Two-smple T-test to test wether the callback successes for the black and the white have the similar mean. The sample size is 4870, which is well greater than 30. Plus, the samples are based on different individuals. Therefore, the CLT would apply for this dataset. 

# 2
What are the null and alternate hypotheses?

# Solution:
## Null: $\mu_{white\_\\callback} - \mu_{black\_\\callback} = 0$
## Hypothesis: $\mu_{white\_\\callback} - \mu_{black\_\\callback} \neq 0$

# 3
Compute margin of error, confidence interval, and p-value.

In [17]:
p_value = stats.ttest_ind(White_calls.call, Black_calls.call, equal_var=False)[1]
sample_mean_diff = np.mean(White_calls.call) - np.mean(Black_calls.call)
t_critical = stats.t.ppf(q = 0.975, df = White_calls.shape[0] + Black_calls.shape[0] - 2)

#Get the std of the two samples and compute the common std
S_w = np.std(White_calls.call)
S_b = np.std(Black_calls.call)
S_total = np.sqrt(((White_calls.shape[0] - 1)*S_w**2 + (Black_calls.shape[0] - 1)*S_b**2)/
           (White_calls.shape[0] + Black_calls.shape[0] - 2))

#Combined N
N_total = np.sqrt(1/(White_calls.shape[0] + Black_calls.shape[0]))

margin_of_error = t_critical * S_total * N_total
confidence_interval = (sample_mean_diff - margin_of_error, 
                       sample_mean_diff + margin_of_error)   

print("Margin of Error:", margin_of_error)
print("Confidence Interval:", confidence_interval)
print("P-value:", p_value)

Margin of Error: 0.00762939897551
Confidence Interval: (0.024403455879553743, 0.039662253830567415)
P-value: 3.94294151365e-05


# 4
Write a story describing the statistical significance in the context of the original problem.

# Conclusion:
Based on the statistical testing results, it suggests that the callback successes for the blacks and whites are different at the 5% level since the p-value for the two-smaple T-test is less than 0.05. The confidence interval for the difference between the average callback successes for the whites and blacks ($\bar{x}_{white} - \bar{x}_{black}$) is (0.0244, 0.0397), which doesn't contain 0, i.e. the given sample indicates that the whites have higher average callbacks than the blacks.  

# 5
Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

# Conclusion
The analysis above doesn't gurentee that race/name is the most important factor in callback success since none of the other factors have been studied yet. 

To futher  and better understand the cause and effect relationship between the variabels and callback success, the correlation coefficients between each variable and callback should be calculated and the corresponding correlation test should be performed.